### PR TITLE
Adding 1-d Logarithmic and Exponential models

### LINKED ISSUE #7792: [Feature Request] Exponential1D model
I found there are polynomial models but not exponential models in [astropy's modeling](http://docs.astropy.org/en/stable/modeling/index.html).

What about including `Exponential1D` and `Exponential2D`?

### PR DESCRIPTION
**Pull request by jbkalmbach**
### Description

Resolves #7792. This PR includes class definitions along with documentation and unit tests for Exponential1D, Logarithmic1D.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! Unfortunately, there is a problem with the commit history. Please get rid of the merge commit and also resolve the conflicts.

**Comment by jbkalmbach** [comment]
Ok, I think I did it right.

**Comment by pllim** [comment]
Great, commit history looks good now. Please add a change log and let the full CI run.

File: astropy/modeling/tests/test_models.py [hunk 1]
```diff
@@ -768,3 +769,16 @@ def test_parameter_inheritance():
     assert b.h == 5
     assert b.f ==3
     assert b.f.fixed == True
+
+
+class ExponentialAndLogarithmicTester:
```
Comment (pllim) [inline astropy/modeling/tests/test_models.py]: There might be a better way to test this, but I'll defer to the sub-package maintainer.

File: astropy/modeling/functional_models.py [hunk 2]
```diff
@@ -2640,3 +2640,80 @@ def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
...
+    x_0 = Parameter(default=1)
+
+    @staticmethod
+    def evaluate(x, amplitude, x_0):
+        return amplitude * np.exp(x / x_0)
+
+    @staticmethod
+    def fit_deriv(x, amplitude, x_0):
+        d_amplitude = np.exp(x / x_0)
+        d_x_0 = -amplitude * np.exp(x / (x_0 ** 2))
```
Comment (nden) [inline astropy/modeling/functional_models.py]: Could you double check this derivative. It should be with respect to `x_0`.

**Comment by nden** [comment]
@jbkalmbach Do you have time to finish this before Friday? If not, do you mind if I push to your branch and try to get this PR in the 4.0 release?

**Comment by jbkalmbach** [comment]
Yeah, I will make some time to work on it this evening.

File: astropy/modeling/functional_models.py (hunk 2 above)
Comment (jbkalmbach) [reply astropy/modeling/functional_models.py]: Ah yep, you are correct. Changing that.

File: CHANGES.rst [hunk 3]
```diff
@@ -433,6 +433,8 @@ astropy.io.votable
 astropy.modeling
 ^^^^^^^^^^^^^^^^
 
+- Added Exponential1D and Logarithmic1D models for fitting [#9351]
```
Comment (karllark) [inline CHANGES.rst]: This is in the bugfix section.  Probably should be in the new features section.

File: astropy/modeling/tests/example_models.py [hunk 4]
```diff
@@ -217,6 +217,22 @@
         'x_lim': [0, 10],
         'y_lim': [0, 10],
     }
```
Comment (bsipocz) [inline astropy/modeling/tests/example_models.py]: missing `,` causing syntax error down below.

File: astropy/modeling/tests/example_models.py [hunk 5]
```diff
@@ -217,6 +217,22 @@
...
+
+    Exponential1D: {
+        'parameters': [1, 1],
+        'x_values': [0, 0.5, 1],
+        'y_values': [1, np.sqrt(np.e), np.e]
+        'x_lim': [0, 2],
+        'integral': (np.e**2 - 1.),
+    }
```
Comment (bsipocz) [inline astropy/modeling/tests/example_models.py]: here, too

**Comment by nden** [comment]
@jbkalmbach This looks ready to be merged. Do you mind squashing all commits?

**Comment by bsipocz** [comment]
> Do you mind squashing all commits?

it shouldn't be all, as there are two different authors, but the last 6 of try and errors with the CI could indeed be in a single one.

**Comment by nden** [comment]
Ah, sorry, I think I just created a conflict. Please rebase again.

**Comment by bsipocz** [comment]
No need to wait for CI again, this was passing before the squash/rebase and doesn't none of the recent changes are affecting it. I'm cancelling the CIs.

**Comment by nden** [comment]
@bsipocz I missed the fact there was a second author.

**Comment by nden** [comment]
Thanks @jbkalmbach !

**Pull request merged**